binderhub/redirect.py

```diff
@@ -1,16 +1,91 @@
 """
 Handler for URL redirection
 """
-from tornado import web, gen
+import base64
+import json
+import random
+import string
+import uuid
+from urllib.parse import quote
+
+from tornado.log import app_log
+from tornado import web
+from tornado.httpclient import AsyncHTTPClient, HTTPRequest, HTTPError
+from tornado.httputil import url_concat
 
 from .base import BaseHandler
 
+# Add a random lowercase alphanumeric suffix to usernames to avoid collisions
```

Comment:
Perhaps:
```
# Add a random lowercase alphanumeric suffix to usernames to avoid collisions
# Set characters which may be used to generate a suffix
SUFFIX_CHARS...
# Set length of suffix. Number of combinations = SUFFIX_CHARS**SUFFIX_LENGTH = 36**8 ~= 2**41
SUFFIX_LENGTH...
```